src/sema/expression.rs

```diff
@@ -4522,6 +4522,21 @@ fn member_access(
         diagnostics,
         resolve_to,
     )? {
+        // Acccessing constant values on contract names is only valid for base or library contracts
+        if let pt::Expression::Variable(namespace) = e {
+            if symtable.find(&namespace.name).is_none() {
+                if let (Some(base), Some(contract)) = (
+                    ns.resolve_contract(context.file_no, namespace),
+                    context.contract_no,
+                ) {
+                    if !is_base(base, contract, ns) && !ns.contracts[base].is_library() {
+                        diagnostics
+                            .push(Diagnostic::error(e.loc(), "member access for constant values via contract name is only valid for base contracts".into()));
+                        return Err(());
+                    }
+                }
+            }
+        }
```

Comment:
I am not sure about this, for two reasons.

First of all, the code is not right because:
 - any restrictions should be handled in `fn contract_constant()`
 - contracts can be imported via an import object, in which case `e` can refer to `a.b.c`, i.e. `pt::Expression::Namespace("a", pt::Expression::Variable("b"))`. The code does not handle this case

The other reason is - why is it not permitted to use a constant in a contract? I know that solc does not permit it, but I'm not really sure why they done this.